Utilities/types.h

```diff
@@ -21,20 +21,34 @@
 #define IS_BE_MACHINE 0
 
 #ifdef _MSC_VER
-#define ASSUME(cond) __assume(cond)
+
+#define ASSUME(cond) __assume(cond) // MSVC __assume ignores side-effects
```

Comment:
Old problem, but most macro should use ... and __VA_ARGS__ instead of a single named arg.